### ISSUE TITLE
Can AA sequences be used in DADA?

### ISSUE THREAD

**Issue by GOlysse** [benjjneb/dada2#563]
Hello,
I have an inquiry in regards to adapting the DADA2 package for usage of protein sequences. After you learn the error rates for the nucleotide sequences, if you were to translate the sequences to protein sequences would you be able to continue with the following dereplication step?

I am using a protein coding gene as my gene marker for PCR and so the variances that are observed are many, which is to be expecting with this type of dataset, but I know that if I could use AA sequences the variation would decrease and I would thus have a table of single amino acid variances.

Any suggestions?

**Comment by benjjneb** [comment]
>  if you were to translate the sequences to protein sequences would you be able to continue with the following dereplication step?

It will dereplicate, but subsequent steps will break as they are expecting DNA sequences.

But, why not run through the DADA2 workflow on the DNA, and then translate back to AA at the end, once you have the error-corrected table of DNA sequences? That should work fine.

**Comment by GOlysse** [comment]
So, my idea was that I would follow to tutorial through the error rate step, then translate the sequences to AA, then dereplicate the AA sequences. 

The script for dereplication is what I would really like to apply to the AA sequences.

I am concerned that once I export the nucleotide sequences for translation, I wouldn’t be able to import them back into DADA without losing the sequence IDs etc

Georgina Olysse

**Comment by GOlysse** [comment]
My apologies, yes I was referring to the derepFastq step. Once I translate the nucleotide sequences to AA sequences there are bound to be AA sequences that are identical and so I wanted to take advantage of the derepFastq step to merge identical AA sequences as well. 

Georgina Olysse

**Comment by benjjneb** [comment]
> Once I translate the nucleotide sequences to AA sequences there are bound to be AA sequences that are identical and so I wanted to take advantage of the derepFastq step to merge identical AA sequences as well.

Yes, but `derepFastq` isn't the right tool for that, it's a specialized tool for reading in (and dereplicating) DNA fastq files.

If you go through to the sequence table, then translate the column names of the sequence table, you can collapse columns (AA sequences) that are then identical at that point. I think `collapseNoMismatch(st.translated)` should do the trick, but if not a couple basic R commands would do it (perhaps `t(rowsum(t(st.translated), colnames(st.translated)))`).

**Comment by GOlysse** [comment]
Thanks!

Georgina Olysse
